sr_wear_logger/src/sr_wear_logger/sr_wear_logger.py

```diff
@@ -0,0 +1,239 @@
+            if bool(re.match(pattern, file)):
+                os.remove(cls.path_to_test_folder + "/" + file)
+
+    def check_parameters(self):
+        check_serial = self._hand_serial is not None and self._hand_serial != ""
+        check_hand_id = self._hand_id == "rh" or self._hand_id == "lh"
+        check_aws = self._aws_save_period is not None and type(self._aws_save_period) == int
+        check_local = self._local_save_period is not None and type(self._local_save_period) == int
+        return check_serial and check_hand_id and check_aws and check_local
+
+    def run(self):
+        if self.check_parameters():
+            self.aws_manager = AWS_Manager()
+            self._verify_log_structure()
+            self.t_local = rospy.Timer(rospy.Duration(self._local_save_period), self._save_data_localy)
+            self.t_aws = rospy.Timer(rospy.Duration(self._aws_save_period), self._upload_to_AWS)
+            rospy.Subscriber('/joint_states', JointState, self._callback)
+            rospy.loginfo("SrWearLogger initialized!")
+            self._running = True
+        else:
+            rospy.logwarn("Can't run SrWearLogger. Wrong parameters!")
+            rospy.signal_shutdown("")
+
+    def _verify_log_structure(self):
+        if not os.path.exists(self._log_file_path):
+            os.makedirs(self._log_file_path)
+        if os.path.exists(self._log_file_path + self._log_file_name):
+            self._sync_and_update_log_file()
+        else:
+            self._generate_log_file()
+
+    def _sync_and_update_log_file(self):
+        if os.path.exists(self._log_file_path + self._log_file_name):
+            shutil.copy(self._log_file_path + self._log_file_name, self._log_file_path + "/wear_data_local.yaml")
+
+            local_file_path = self._log_file_path + "/wear_data_local.yaml"
+            aws_file_path = self._log_file_path + "/wear_data.yaml"
+
+            if self._download_from_AWS():
+                try:
+                    with open(local_file_path, 'r') as f_local, open(aws_file_path, 'r') as f_aws:
+                        data_local = yaml.load(f_local, Loader=yaml.SafeLoader)
+                        data_aws = yaml.load(f_aws, Loader=yaml.SafeLoader)
+                        self._complete_data = self._update_with_higher_values(data_local, data_aws)
+
+                    if self._complete_data is not None:
+                        self._current_values = self._complete_data['total_angles_[rad]']
+                        self._current_time = self._complete_data['total_time_[s]']
+                        self._save_data_localy()
+                        self._upload_to_AWS()
+
+                except Exception as e:
+                    rospy.loginfo("No local/AWS log file found!")
+
+            if os.path.exists(self._log_file_path + "/wear_data_local.yaml"):
+                os.remove(self._log_file_path + "/wear_data_local.yaml")
+
+        else:
+            rospy.logwarn("Log file doesn't exist!")
```

Comment:
Added rospy.get_name() for all loginfo() in the code.
In case of failed download there's a logwarn from the AWS_Manager class that will print the filename.
```
 try:
                self._client.upload_file(self.file_full_path, bucket_name, self.aws_path)
                uploadSucceded = True
                rospy.loginfo("File upload succeded.")
            except Exception as e:
                rospy.loginfo("File upload failed" + str(e))
```